paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java

```diff
@@ -0,0 +1,102 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.nms.NMSHandler;
+import com.denizenscript.denizen.nms.NMSVersion;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.MapTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.destroystokyo.paper.SkinParts;
+import com.destroystokyo.paper.event.player.PlayerClientOptionsChangeEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class PlayerClientOptionsChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player client options change
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Player Always.
+    //
+    // @Triggers when a player changes their client options.
+    //
+    // @Context
+    // <context.server_listings_enabled> returns an ElementTag(Boolean) of whether the player has server listings enabled. Available only on MC 1.19+.
+    // <context.chat_visibility> returns the player's chat visibility.
+    // <context.locale> returns the player's locale.
+    // <context.main_hand> returns an ElementTag(Boolean) of whether the player's main hand has changed.
+    // <context.skin_parts> returns a MapTag of whether the player's skin parts are enabled or not. For example: [cape=true;jacket=false;left_sleeve=true;right_sleeve=false;left_pants=true;right_pants=false;hat=true]
+    // <context.view_distance> returns the player's view distance.
+    // <context.server_listings_changed> returns whether the player's server listings have changed. Available only on MC 1.19+.
+    // <context.chat_colors> returns whether the player has chat colors enabled.
+    // <context.chat_colors_changed> returns whether the player's chat colors have changed.

```

Comment:
This (and a few other `_changed` meta) seem a bit unclear to me - it's not that the client has a chat color and that changes, it's whether the client allows chat colors that changes